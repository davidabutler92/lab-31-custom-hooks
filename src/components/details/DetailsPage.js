import React from 'react';
import PropTypes from 'prop-types';

function DetailsPage({ character }) {
  return (
    <div>
      <img src={character.image} alt={character.name} />
      <h1>name: {character.name}</h1>
      <p>status: {character.status}</p>
      <p>species: {character.species}</p>
    </div>
  );
}

DetailsPage.propTypes = {
  character: PropTypes.shape({
    image: PropTypes.string,
    name: PropTypes.string,
    status: PropTypes.string,
    species: PropTypes.string,
  }),
};

export default DetailsPage;
