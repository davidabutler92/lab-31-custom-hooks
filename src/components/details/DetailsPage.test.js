import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DetailsPage from './DetailsPage';

describe('DetailsPage component', () => {
  afterEach(() => cleanup());
  it('renders a details of a single character', () => {
    const { asFragment } = render(
      <DetailsPage
        character={{
          id: 1,
          image: 'some image',
          name: 'some name',
          status: 'some status',
          species: 'some species',
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
