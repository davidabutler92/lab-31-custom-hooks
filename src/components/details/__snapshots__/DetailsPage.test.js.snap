// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component renders a details of a single character 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="some name"
      src="some image"
    />
    <h1>
      name: some name
    </h1>
    <p>
      status: some status
    </p>
    <p>
      species: some species
    </p>
  </div>
</DocumentFragment>
`;
